lib/kernel/src/dist_util.erl

```diff
@@ -695,8 +695,14 @@ send_status(#hs_data{socket = Socket, other_node = Node,
 	_ -> 
 	    true
     end.
-    
-    
+
+%% Constant time comparison between two binaries of the same size.
+secure_compare(B1, B2) ->
+    secure_compare(B1, B2, 0) =:= 0.
+secure_compare(<<H1, T1/binary>>, <<H2, T2/binary>>, Acc) ->
+    secure_compare(T1, T2, Acc bor (H1 bxor H2));
+secure_compare(<<>>, <<>>, Acc) ->
+    Acc.
```

Comment:
This comparison should be time-insensitive when considering the local "correct" cookie, so the comparison time should be tied directly to the remote "test" cookie.  Currently, this source code is not handling different lengths.  An example of what I am describing is at https://github.com/CloudI/CloudI/blob/develop/src/lib/cloudi_core/src/cloudi_string.erl#L415-L422